BranchMetrics/ios-branch-deep-linking-attribution
# SDK-2309 cache link data when initialization is deferred

## Reference
SDK-2309 Cache link data when initialization is deferred

## Summary
On React Native we had a caching mechanism to store links until the JS layer was ready. This is not working properly when using the optional deferred init on iOS. This moves the caching to the native iOS layer.

## Motivation
Fix cold link open when using deferred init on iOS with React Native

## Type Of Change

## Testing Instructions
Enable deferred init with the `branch.json`
Add a delayed `notifyNativeToInit`
Install app, but make sure it is fully closed
Open the app with a universal link

Without this change the link will be ignored.
With this change the link will be used with the deferred init.

Also need to test this on React Native.

cc @BranchMetrics/saas-sdk-devs for visibility.
